### PR TITLE
Refactoring lat long

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [x] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- changed format latitude and longitude > array of floats  
- changed the response of `post` and `patch` method to return full objects
- refactored responses that were not JSON:API compliant

## Summary of the change.

🐞 If it is a bug or issue fix, describe the issue that was fixed.

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/resources/list_shopping_lists.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def get(self):
       for item in lst.items:
         total += item.quantity
       list_info['item_count'] = total
-      list_info['lat_long'] = "{},{}".format(lst.at_risk_user.store[0].latitude, lst.at_risk_user.store[0].longitude)
+      list_info['latitude_longitude'] = [float(lst.at_risk_user.store[0].latitude), float(lst.at_risk_user.store[0].longitude)]
```
Comment (adumortier) [inline trackbasket_be/resources/list_shopping_lists.py:25]: reformated `lat` and `long` to array of floats [`lat`, `long`]

File: trackbasket_be/resources/shopping_list.py [hunk 2]
```diff
@@ -21,7 +22,12 @@ def post(self,id):
       for item in request_data["items"]:
         db.session.add(Item(shopping_list_id=shopping_list.id, **item)) 
       db.session.commit()
-      return {'message': 'new shopping list created for user #{}'.format(id)}, 200
+      return {'data': {  "id": "shoppinglist","attributes":
+            { "name": store.name, "address": store.address, "city": store.city, "state": store.state, "zipcode": store.zipcode,
+              "storeId": store.location_id, "latitude_longitude": [float(store.latitude), float(store.longitude)], "status": shopping_list.status,
+              "created_date": shopping_list.created_date.strftime("%d/%m/%Y %H:%M:%S"),
+              "items": [item.json() for item in shopping_list.items] } } }, 200
+      # return {'message': 'new shopping list created for user #{}'.format(id)}, 200
```
Comment (adumortier) [inline trackbasket_be/resources/shopping_list.py:30]: need to remove this comment

File: trackbasket_be/resources/volunteer.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ def post(self, id):
     volunteer = Volunteer(name=request_data['name'], phone_number=request_data['phone_number'], volunteer_id=id)
     db.session.add(volunteer)
     db.session.commit()
-    return {'volunteer': request_data}, 201
+    return {'data': { 'id': 'volunteer', 'attributes': {'id': volunteer.volunteer_id, 'name': volunteer.name, 'phone number': volunteer.phone_number} } }, 201
```
Comment (adumortier) [inline trackbasket_be/resources/volunteer.py:19]: returns full volunteer object

File: trackbasket_be/resources/volunteer.py [hunk 4]
```diff
@@ -31,6 +31,10 @@ def patch(self, id):
       return {'data': { 'id': 'volunteer', 'attributes': {'id': volunteer.volunteer_id, 'name': volunteer.name, 'phone number': volunteer.phone_number} } }, 200
 
   def delete(self, id):
-    db.session.query(Volunteer).filter_by(volunteer_id=id).delete()
-    db.session.commit()
+    volunteer = db.session.query(Volunteer).filter_by(volunteer_id=id)
+    if volunteer.first() is None:
+      return {'data': { 'id': 'volunteer', 'attributes': {'error': "volunteer does not exist"} } }, 400
```
Comment (mronauli) [inline trackbasket_be/resources/volunteer.py:36]: nice catch on the edge case!

File: trackbasket_be/resources/shopping_list.py [hunk 5]
```diff
@@ -35,24 +41,34 @@ def get(self,id):
...
       for item in request_data["items"]:
         db.session.add(Item(shopping_list_id=shopping_list.id, **item)) 
       db.session.commit()
-      return {'message': 'shopping list of user {} was updated'.format(id)}, 200
+      
+      return {'data': {  "id": "shoppinglist","attributes":
+            { "name": store.name, "address": store.address, "city": store.city, "state": store.state, "zipcode": store.zipcode,
+              "storeId": store.location_id, "latitude_longitude": [float(store.latitude), float(store.longitude)], "status": shopping_list.status,
+              "created_date": shopping_list.created_date.strftime("%d/%m/%Y %H:%M:%S"),
+              "items": [item.json() for item in shopping_list.items] } } }, 200
```
Comment (mronauli) [inline trackbasket_be/resources/shopping_list.py:70]: can probably move into the shopping list model to clean resource up a bit

**Pull request merged**